### ISSUE TITLE
Non-deterministic behavior when using threading and torch agents

### ISSUE THREAD

**Issue by omardrwch** [rlberry-py/rlberry#64]
The output of the code below is only reproducible when `parallelization='process'`, and not when `parallelization='thread'`.

This seems to happen only with torch agents. In each thread, `AgentManager` calls `set_external_seed()`, which sets the seed of PyTorch, so I guess the problem is that PyTorch is sharing seeds among threads, and we get a non-deterministic behavior inherent to multithreading. 

I don't think it is necessarily something to be fixed, but it's nice to be aware of it.

```python

import numpy as np
from rlberry.envs.benchmarks.ball_exploration import PBall2D
from rlberry.agents.torch.a2c import A2CAgent
from rlberry.manager import AgentManager, plot_writer_data, evaluate_agents
from rlberry.seeding import set_external_seed


if __name__ == '__main__':
    set_external_seed(123)

    # --------------------------------
    # Define train and evaluation envs
    # --------------------------------
    train_env = (PBall2D, dict())
    eval_env = (PBall2D, dict())

    # -----------------------------
    # Parameters
    # -----------------------------
    N_EPISODES = 250
    GAMMA = 0.99
    HORIZON = 50

    params_a2c = {"gamma": GAMMA,
                  "horizon": HORIZON,
                  "learning_rate": 0.0003}

    eval_kwargs = dict(eval_horizon=HORIZON, n_simulations=20)

    a2c_stats = AgentManager(
        A2CAgent,
        train_env,
        fit_budget=N_EPISODES,
        init_kwargs=params_a2c,
        eval_kwargs=eval_kwargs,
        n_fit=4,
        seed=123,
        parallelization='thread')

    agent_manager_list = [a2c_stats]

    for st in agent_manager_list:
        st.fit()

    # learning curves
    plot_writer_data(agent_manager_list,
                     tag='episode_rewards',
                     preprocess_func=np.cumsum,
                     title='cumulative rewards',
                     show=False)

    # compare final policies
    output = evaluate_agents(agent_manager_list)
    print(output)

    for st in agent_manager_list:
        st.clear_output_dir()
```
